### PR TITLE
[stable/datadog] Allow custom agent configuration

### PR DESCRIPTION
**Pull request by gaida**
#### What this PR does / why we need it:
We need to have the ability to provide custom configuration to the datadog agent.

In the current state we have to rely on either the default configuration of the agent baked into the container, or configuration through a hard coded configmap located in the chart. The configmap could potentially be changed after successful deployment, but that requires a restart of the pods and that pattern is generally not nice for automation.

This PR solves this issue by providing the ability to use all custom agent configuration through the variable `.Values.daemonset.customAgentConfig`. Backwards compatibility is provided by using the default config inside the configmap if that variable is not set.

#### Which issue this PR fixes
-/-

#### Special notes for your reviewer:
-/-

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by gaida** [comment]
Hi @mfpierre, @hkaj,

sorry for the bump, but do you know when you'll be able to take a look at this? We want to run this code, but it would be easiest for us if this change made it upstream.

Thanks!
- Martin

**Comment by gaida** [comment]
Sorry for the highlight of all approvers, but this request has gone without acknowledge for 1,5 months and is about to be closed due to being stale. It would be great if somebody could take a quick look, it's a small change and we're currently running it - so it's tested.

Thanks!

@hkaj
@irabinovitch
@charlyf
@mfpierre
@clamoriniere
@xlucas

File: stable/datadog/README.md [hunk 1]
```diff
@@ -286,6 +286,8 @@ helm install --name <RELEASE_NAME> \
 | `daemonset.containers.traceAgent.resources.limits.memory`        | Memory resource limits for the trace-agent container                                   | `200Mi`                                       |
 | `daemonset.containers.traceAgent.resources.requests.memory`      | Memory resource requests for the trace-agent container                                 | `200Mi`                                       |
 | `daemonset.priorityClassName`            | Which Priority Class to associate with the daemonset                                      | `nil`                                       |
+| `daemonset.useConfigMap`                 | Configures a configmap to provide the agent configuration                                 | `false`                                     |
```
Comment (gaida) [inline stable/datadog/README.md:323]: fyi: this is an existing value that wasn't documented yet. Not part of the code change here.

**Comment by irabinovitch** [comment]
@gaida Let me try to line up a review for this over the next few days.

**Comment by CharlyF** [comment]
@gaida - Apologies for the delay here. This is a change that would be helpful, thanks for taking the time to making it.
Could you please rebase ? I'll validate and approve subsequently.

**Comment by gaida** [comment]
@CharlyF @irabinovitch thanks for the response and sorry for the delay from my side this time. I've rebased + bumped the version. Please take a look, thank you.

/cc @hkaj @mfpierre

**Comment by gaida** [comment]
bumped version.

and sorry for the bump, but it would be great if someone could take a look at this before this goes stale again. It's a tiny change that we've been running for a long time and it would be great if this could go into upstream. Thank you!

@hkaj
@irabinovitch
@CharlyF
@mfpierre
@clamoriniere
@xlucas

**Comment by CharlyF** [comment]
All good.
/lgtm

**Comment by gaida** [comment]
Awesome, thanks for the quick approval @CharlyF !

**Pull request merged**